File: leetcode/js/48. Rotate Image.js
function rotate(arr) {
  const n = arr.length;
  for (let i = 0; i < n / 2; i++) {
    for (let j = 0; j < Math.ceil(n / 2); j++) {
      let temp = arr[i][j];
      arr[i][j] = arr[n - 1 - j][i];
      arr[n - 1 - j][i] = arr[n - 1 - i][n - 1 - j];
      arr[n - 1 - i][n - 1 - j] = arr[j][n - 1 - i];
      arr[j][n - 1 - i] = temp;
    }
  }
}

const arr = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],
];

console.log(arr);
console.log("#############");
rotate(arr);
console.log(arr);

File: leetcode/java/48. Rotate Image.java
import java.io.*;

public class Main {

    public static PrintStream ps = System.out;

    public static void rotate(int[][] matrix) {
        int n = matrix.length;
        for (int i = 0; i < n / 2; i++) {
            for (int j = 0; j < Math.ceil(((double) n) / 2.); j++) {
                int temp = matrix[i][j];
                matrix[i][j] = matrix[n - 1 - j][i];
                matrix[n - 1 - j][i] = matrix[n - 1 - i][n - 1 - j];
                matrix[n - 1 - i][n - 1 - j] = matrix[j][n - 1 - i];
                matrix[j][n - 1 - i] = temp;
            }
        }
    }

    public static void display(int[][] arr) {
        for (int[] i : arr) {
            for (int j : i) {
                ps.print(j + " ");
            }
            ps.println("");
        }
    }

    public static void main(String... s) {
        int arr[][] = { { 1, 2, 3 }, { 4, 5, 6 }, { 7, 8, 9 } };
        display(arr);
        rotate(arr);
        ps.println("################");
        display(arr);
    }
}